src/behavior-not-considered-unsafe.md

```diff
@@ -37,3 +37,18 @@ See [RFC 560] for error conditions, rationale, and more details about
 integer overflow.
 
 [RFC 560]: https://github.com/rust-lang/rfcs/blob/master/text/0560-integer-overflow.md
+
+##### Logic errors
+
+Some safe functionality may impose extra logic constraints that cannot be
+checked at neither compile-time nor runtime, for different reasons. If a user
+breaks such a constraint, the behavior is unspecified but will not result in
```

Comment:
```suggestion
breaks such a constraint, the behavior may be unspecified but will not result in
```

It's not unreasonable to possibly specify behavior for code that breaks contracts.

Reply:
I see your point. Hmm... when I was writing this, I was thinking about this as a definition, i.e. that if somewhere else a "logic error" is referenced, then it means the behavior is unspecified. Thus if you actually specify what happens when some constraints are broken for some function, then you shouldn't use the term "logic error" (or whatever we chose for that term). So, for instance, an `unwrap()` panicking wouldn't be regarded as a "logic error".

But I agree with you, because this is not actually written as a definition (it uses "logic error" rather than defining it) and the file isn't meant to be a list of definitions either. Plus giving a very specific meaning to a term as common as "logic error" would create confusion. Nevertheless, I wonder if such a list of semi-formal definitions would be a good idea (similar to what I comment in the OP).